dashboard/src/components/DeploymentFormBody/Differential.tsx

```diff
@@ -1,37 +1,57 @@
-import * as jsdiff from "diff";
-import * as Diff2Html from "diff2html";
-import React from "react";
+import ReactDiffViewer, { DiffMethod } from "react-diff-viewer";
+import { useSelector } from "react-redux";
+import { SupportedThemes } from "shared/Config";
+import { IStoreState } from "shared/types";
 
-import "diff2html/bundles/css/diff2html.min.css";
 import "./Differential.css";
 
 export interface IDifferentialProps {
-  title: string;
   oldValues: string;
   newValues: string;
   emptyDiffText: string;
 }
 
-class Differential extends React.Component<IDifferentialProps> {
-  public render = () => {
-    const { oldValues, newValues, title, emptyDiffText } = this.props;
-    const sdiff = jsdiff.createPatch(title, oldValues, newValues);
-    const outputHtml = Diff2Html.html(sdiff, {
-      inputFormat: "diff",
-      showFiles: false,
-      matching: "lines",
-      maxLineSizeInBlockForComparison: 20,
-    } as Diff2Html.Diff2HtmlConfig);
-    return (
-      <div className="diff deployment-form-tabs-data">
-        {oldValues === newValues ? (
-          <span>{emptyDiffText}</span>
-        ) : (
-          <div dangerouslySetInnerHTML={{ __html: outputHtml }} />
-        )}
-      </div>
-    );
+function Differential(props: IDifferentialProps) {
+  const { oldValues, newValues, emptyDiffText } = props;
+  const {
+    config: { theme },
+  } = useSelector((state: IStoreState) => state);
+
+  // Modify colors to match the Advanced Tab theme
+  const newStyles = {
+    variables: {
+      dark: {
+        // gutter
+        gutterColor: "#d0edf7",
+        gutterBackground: "#01313f",
+        gutterBackgroundDark: "#01313f",
+        addedGutterColor: "#d0edf7",
+        removedGutterColor: "#d0edf7",
+        // background
+        diffViewerBackground: "#002B36",
+        emptyLineBackground: "#002B36",
+        // fold text
+        codeFoldContentColor: "white",
+      },
+    },
```

Comment:
Mmm, is it required to harcode the colors in the tsx? Isn't there any css approach? I guess no, but wanted to confirm anyway

Reply:
the documented way of changing the style is through this `styles` object: https://github.com/praneshr/react-diff-viewer#overriding-styles

It may be possible to achieve the same through CSS but it's not documented. Also, while they should maintain the compatibilty of this API (the styles object) they don't need to do so for the CSS which means that if we use CSS only, it may break in future minor releases.